src/test/kotlin/com/atlassian/performance/tools/ssh/api/SshTest.kt

```diff
@@ -56,14 +56,24 @@ class SshTest {
         SshContainer().useSsh { sshHost ->
             installPing(sshHost)
 
-            val fail = sshHost.runInBackground("nonexistent-command")
-            val failResult = fail.stop(Duration.ofMillis(20))
+            repeat(1000) {
+                val fail = sshHost.runInBackground("nonexistant-command")
+                sshHost.waitForAllProcessesToFinish("nonexistant-command", Duration.ofMillis(100))
```

Comment:
Timeout with `Duration.ofMillis(50)` wasn't enough. I had a run where in 1000 repeats it timed out.

I believe that those kind of timeouts are inevitable when we are working with OS. There could always be some huge lag spike where this and any other processes with timeout would just fail.

Anyway it's better to timeout on a ssh command where an exception explicitly says that it was a timeout then when we fail because we didn't get a response. In the former it's obvious that we can just increase the timeout to reduce the likelyhood of the flakiness.

Reply:
> I believe that those kind of timeouts are inevitable when we are working with OS.

Yeah, process scheduling can introduce random delays, but the delay distribution is not open-ended. It will never take 1 hour.

> Anyway it's better to timeout on a ssh command where an exception explicitly says that it was a timeout then when we fail because we didn't get a response. In the former it's obvious that we can just increase the timeout to reduce the likelyhood of the flakiness.

Exactly :heart: